selfhost/typechecker.jakt

```diff
@@ -3247,7 +3265,7 @@ struct Typechecker {
                     .check_types_for_compat(
                         lhs_type_id: checked_block.yielded_type.value()
                         rhs_type_id: type_
-                        generic_enferences: generic_inferences
+                        generic_inferences: generic_inferences
```

Comment:
Unrelated change. (Also, the fact that this worked before is probably a bug on its own!)